<launch>
    <arg name="is_sim" value="1"/>
    <arg name="traj_gen" value="1"/>
    <arg name="target_timestep" value="0.5"/>
    <arg name="num_legs" value="3"/>

    <include file="$(find gtddp_drone)/launch/constants.launch">
        <arg name="ddp_select" value="pursuit"/>

        <!-- DDP Hyperparameters -->
        <arg name="num_time_steps" value="2001"/>
        <arg name="num_iterations" value="75"/>
        <arg name="num_long_legs" value="30"/>
        <arg name="short_iterations" value="15"/>
        <arg name="dt" value="0.001"/>
        <arg name="learning_rate" value="0.03"/>

        <!-- AR.Drone System Parameters -->
        <arg name="num_states" value="24"/>
        <arg name="mass" value="0.436"/>
        <arg name="length" value="0.19"/>
        <arg name="Ixx" value="0.0045"/>
        <arg name="Iyy" value="0.0051"/>
        <arg name="Izz" value="0.0095"/>

        <!-- Initial conditions for u_traj -->
        <arg name="u0_hover" value="5"/>
        <arg name="v0_hover" value="5"/>

        <!-- Cost Function Parameters -->
        <arg name="Ru" value="0.1" />
        <arg name="Rv" value="10" />
        <arg name="Q1" value="1000" />
        <arg name="Q2" value="1000" />
        <arg name="Q3" value="1000" />
        <arg name="Qx_multiplier" value="0" />

        <!-- Constrain the pursuit controls? -->
        <arg name="constrained" default="false" />
    </include>

    <param name="gtddp_optimize_node/is_gen" value="$(arg traj_gen)"/>
    <param name="gtddp_optimize_node/gen_legs" value="$(arg num_legs)"/>
</launch>
